### PR TITLE
Feat: add plop to api actions

### PR DESCRIPTION
**Pull request by szu2**

### CODE REVIEW

**Review by qh4r — APPROVED** [review]

**Comment by barto-tsh** [comment]
please go through: https://github.com/TheSoftwareHouse/react-starter-boilerplate/pull/43/files and add empty lines at the end of files 😁 

image

File: plop-templates/apiActions/apiActions.types.hbs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export type {{pascalCase name}}MutationArguments = {
```
Comment (barto-tsh) [inline plop-templates/apiActions/apiActions.types.hbs:1]: it'd be also good to add types for errors from API 😁

File: plopfile.js [hunk 2]
```diff
@@ -86,6 +93,68 @@ const reactComponentGenerator = () => ({
...
+      {
+        type: 'modify',
+        path: 'src/ui/index.ts',
+        pattern: 'export',
+        templateFile: 'plop-templates/component/Component.index.hbs',
+        skip() {
+          if (data.baseDir !== 'ui') {
+            return 'Skipped adding export to src/ui/index.ts';
+          }
+          return;
```
Comment (barto-tsh) [inline plopfile.js:105]: I think this `return` is not necessary here 😁 omitting it does the same

**Review by barto-tsh — APPROVED** [review]

**Comment by barto-tsh** [comment]
good job with this 😁

**Comment by szu2** [comment]
> 

You are right to add new lines on some files, but not these two there are on the screen :D I did it, thanks.

**Comment by barto-tsh** [comment]
> > 
> 
> You are right to add new lines on some files, but not these two there are on the screen :D I did it, thanks.

ahh right 🤦 those are templates 😅

**Review by barto-tsh — APPROVED** [review]

**Review by szymonchudy — APPROVED** [review]

**Comment by krzysztofkrolikowski** [comment]
You have few places with removed newline on EOF(End Of File). Its generally recommended to use such newline. Could you add them?

File: plop-templates/component/Component.index.hbs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export * from './{{camelCase name}}/{{pascalCase name}}';
+export
```
Comment (krzysztofkrolikowski) [inline plop-templates/component/Component.index.hbs:2]: This lonely export seems to be not necessary. Could you check that?

**Review by qh4r — APPROVED** [review]

**Pull request merged**